scala/scala.bzl

```diff
@@ -289,23 +289,65 @@ DependencyAnalyzerMode: {dependency_analyzer_mode}
         arguments=["--jvm_flag=%s" % flag for flag in ctx.attr.scalac_jvm_flags] + ["@" + argfile.path],
       )
 
+    java_jar = try_to_compile_java_jar(ctx, all_srcjars, java_srcs, implicit_junit_deps_needed_for_java_compilation)
+    return java_jar
 
-def _compile_or_empty(ctx, jars, srcjars, buildijar, transitive_compile_jars, jars2labels):
+
+def try_to_compile_java_jar(ctx, all_srcjars, java_srcs, implicit_junit_deps_needed_for_java_compilation):
+    if not java_srcs and not all_srcjars:
+      return False
+
+    providers_of_dependencies = collect_java_providers_of(ctx.attr.deps)
+    providers_of_dependencies += collect_java_providers_of(implicit_junit_deps_needed_for_java_compilation)
+    scala_sources_java_provider = java_common.create_provider(
+        #the line below means we can't use the provider from java_common.compile
+        #since it will include the full jar when I want to expose only the ijar
+        #on the other hand we can't supply the java_common.compile the scala ijar since it needs the internals
+        compile_time_jars = [ctx.outputs.jar],
+        runtime_jars = [],
+    )
+    providers_of_dependencies += [scala_sources_java_provider]
+
+    full_java_jar = ctx.actions.declare_file(ctx.label.name + "_java.jar")
+
+    provider = java_common.compile(
+                ctx,
+                source_jars = all_srcjars.to_list(),
+                source_files = java_srcs,
+                output = full_java_jar,
+                javac_opts = ctx.attr.javacopts + ctx.attr.javac_jvm_flags,
+                deps = providers_of_dependencies,
+                #exports can be empty since the manually created provider exposes exports
+                #needs to be empty since we want the provider.compile_jars to only contain the sources ijar
+                exports = [],
+                java_toolchain = ctx.attr._java_toolchain,
+                host_javabase = ctx.attr._host_javabase,
+    )
+    return struct(jar = full_java_jar, ijar = provider.compile_jars.to_list().pop())
```

Comment:
is there really a contract that the first item is the ijar for this item? Can we add a comment to that effect?

This jar only includes the java output, not the scala, right?

Reply:
See the comment 5 lines above. Is it clear enough?
Yes.

Reply:
yes that's fine.